frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,484 @@
+    namespace: new URLSearchParams(window.location.search).get('catalogNamespace'),
+    fieldSelector: `metadata.name=${new URLSearchParams(window.location.search).get('pkg')}`,
+    selector: {
+      matchLabels: {
+        catalog: new URLSearchParams(window.location.search).get('catalog'),
+      },
+    },
+    prop: 'packageManifest',
+  },
+];
+
+const ViewInstalledOperatorsButton: React.FC<InstallPageMessageButtonProps> = ({
+  namespace,
+  className,
+}) => (
+  <Button
+    className={className}
+    data-test-id="view-installed-operators-btn"
+    variant="link"
+    component="a"
+    href={`/k8s/ns/${namespace}/${referenceForModel(ClusterServiceVersionModel)}`}
+  >
+    View Installed Operators
+  </Button>
+);
+
+const InstallFailedMessage: React.FC<InstallPageMessageProps> = ({ namespace, csvName, obj }) => {
+  const crdMessage = isCRDAvailable(obj)
+    ? 'The required custom resource will need to be created in the operator once resolved.'
+    : '';
+  const installMessage = `The operator did not install successfully. ${crdMessage}`;
+
+  return (
+    <>
+      <span className="co-clusterserviceversion-install__heading">
+        Operator installation failed
+      </span>
+      <br />
+      <span>{installMessage}</span>
+      <div className="co-clusterserviceversion-install__button_div">
+        <Button
+          variant="primary"
+          component="a"
+          href={`/k8s/ns/${namespace}/${referenceForModel(ClusterServiceVersionModel)}/${csvName}`}
+        >
+          View Error
+        </Button>{' '}
+        <ViewInstalledOperatorsButton namespace={namespace} className={''} />
+      </div>
+    </>
+  );
+};
+
+const InstallNeedsApprovalMessage: React.FC<InstallPageMessageProps> = ({
+  namespace,
+  obj,
+  approve,
+}) => (
+  <>
+    <span className="co-clusterserviceversion-install__heading">Manual approval required</span>
+    <br />
+    <p>
+      Review the manual install plan. Once approved, the following resources will be created in
+      order to satisfy the requirements for the components specified in the plan. Click the resource
+      name to view the resource in detail.
+    </p>
+    <div className="co-clusterserviceversion-install__button_div">
+      <Button variant="primary" onClick={() => approve()}>
+        Approve
+      </Button>
+      <Button
+        className="co-clusterserviceversion-button__left_padding"
+        variant="secondary"
+        component="a"
+        href={`/k8s/ns/${namespace}/${referenceForModel(SubscriptionModel)}/${
+          obj.metadata.name
+        }?showDelete=true`}
+      >
+        Deny
+      </Button>
+      <ViewInstalledOperatorsButton namespace={namespace} className={''} />
+    </div>
+  </>
+);
+
+const CreateCRDButton: React.FC<InstallPageMessageButtonProps> = ({ className, obj, disabled }) => {
+  if (!isCRDAvailable(obj)) {
+    return <></>;
+  }
+  const crObj = safeLoad(getCRDFromObject(obj));
```

Comment:
Why `safeLoad`? Is the annotation YAML intead of JSON? We should be using `JSON.prase` and logging errors.

This logic should be part of `getCRDFromObject` to avoid repeating it.

Reply:
For the examples I have been given, I needed to use the safeload to parse.

Reply:
https://gist.github.com/zherman0/ca4c52b870af9aea9442393505cc4e19
Look at the "operatorframework.io/initialization-resource" annotation.

Reply:
> https://gist.github.com/zherman0/ca4c52b870af9aea9442393505cc4e19
> Look at the "operatorframework.io/initialization-resource" annotation.

That annotation worked fine for me with `JSON.prase`. This is JSON, not YAML. We should be using `JSON.parse` rather than `safeLoad`.

Reply:
nit: I'd call this `cr` or `customerResource` instead of `crObj`. The `Obj` is redundant since every CR is also an object.